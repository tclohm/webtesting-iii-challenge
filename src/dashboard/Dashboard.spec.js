// Test away
import React from "react";
import { render } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';

import Dashboard from "./Dashboard";

test("renders both components in Dashboard", () => {
	const { getByTestId, rerender } = render(<Dashboard/>);
	const displayComp = getByTestId("displayComponent");
	expect(displayComp).toBeInTheDocument();

	rerender(<Dashboard/>);
	const controlComp = getByTestId("controlComponent");
	expect(controlComp).toBeInTheDocument();

});

test("does not render a fake component", () => {
	const { queryByTestId } = render(<Dashboard/>);
	const extraFeatureComp = queryByTestId("extraFeatureComponent");
	expect(extraFeatureComp).not.toBeInTheDocument();
});

test("render control lock button", () => {
	const { getByTestId } = render(<Dashboard/>);
	const button = getByTestId("controlLockButton");
	expect(button).toBeInTheDocument();
});

test("render control door button", () => {
	const { getByTestId } = render(<Dashboard/>);
	const button = getByTestId("controlDoorButton");
	expect(button).toBeInTheDocument();
});

test("dashboard state `closed` and `locked` to initialize as false", () => {
	const locked = false;
	const closed = false;
	const tree = render(<Dashboard locked={locked} closed={closed}/>);
	expect(tree.baseElement).toMatchSnapshot();
})