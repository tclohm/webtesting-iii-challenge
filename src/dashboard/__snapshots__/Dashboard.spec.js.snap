// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard state \`closed\` and \`locked\` to initialize as false 1`] = `
<body>
  <div>
    <div
      class="display panel"
      data-testid="displayComponent"
    >
      <div
        class="led green-led"
      >
        Unlocked
      </div>
      <div
        class="led green-led"
      >
        Open
      </div>
    </div>
    <div
      class="controls panel"
      data-testid="controlComponent"
    >
      <button
        class="toggle-btn"
        data-testid="controlLockButton"
        disabled=""
      >
        Lock Gate
      </button>
      <button
        class="toggle-btn"
        data-testid="controlDoorButton"
      >
        Close Gate
      </button>
    </div>
  </div>
</body>
`;

exports[`dashboard state \`closed\` initializes as false 1`] = `
<body>
  <div>
    <div
      class="display panel"
      data-testid="displayComponent"
    >
      <div
        class="led green-led"
      >
        Unlocked
      </div>
      <div
        class="led green-led"
      >
        Open
      </div>
    </div>
    <div
      class="controls panel"
      data-testid="controlComponent"
    >
      <button
        class="toggle-btn"
        data-testid="controlLockButton"
        disabled=""
      >
        Lock Gate
      </button>
      <button
        class="toggle-btn"
        data-testid="controlDoorButton"
      >
        Close Gate
      </button>
    </div>
  </div>
</body>
`;
